tools/generator/Tests/expected.ji/CovariantReturnTypes/Covariant.Returntypes.CovariantInterfaceImplementation.cs

```diff
@@ -0,0 +1,81 @@
+using Java.Interop;
+
+namespace Covariant.Returntypes {
+
+	// Metadata.xml XPath class reference: path="/api/package[@name='covariant.returntypes']/class[@name='CovariantInterfaceImplementation']"
+	[global::Android.Runtime.Register ("covariant/returntypes/CovariantInterfaceImplementation", DoNotGenerateAcw=true)]
+	public partial class CovariantInterfaceImplementation : global::Java.Lang.Object, global::Covariant.Returntypes.ICovariantInterface {
+
+		internal    new     static  readonly    JniPeerMembers  _members    = new JniPeerMembers ("covariant/returntypes/CovariantInterfaceImplementation", typeof (CovariantInterfaceImplementation));
+		internal static new IntPtr class_ref {
+			get {
+				return _members.JniPeerType.PeerReference.Handle;
+			}
+		}
+
+		public override global::Java.Interop.JniPeerMembers JniPeerMembers {
+			get { return _members; }
+		}
+
+		protected override IntPtr ThresholdClass {
+			get { return _members.JniPeerType.PeerReference.Handle; }
+		}
+
+		protected override global::System.Type ThresholdType {
+			get { return _members.ManagedPeerType; }
+		}
+
+		protected CovariantInterfaceImplementation (IntPtr javaReference, JniHandleOwnership transfer) : base (javaReference, transfer) {}
+
+		// Metadata.xml XPath constructor reference: path="/api/package[@name='covariant.returntypes']/class[@name='CovariantInterfaceImplementation']/constructor[@name='CovariantInterfaceImplementation' and count(parameter)=0]"
+		[Register (".ctor", "()V", "")]
+		public unsafe CovariantInterfaceImplementation ()
+			: base (IntPtr.Zero, JniHandleOwnership.DoNotTransfer)
+		{
+			const string __id = "()V";
+
+			if (((global::Java.Lang.Object) this).Handle != IntPtr.Zero)
+				return;
+
+			try {
+				var __r = _members.InstanceMethods.StartCreateInstance (__id, ((object) this).GetType (), null);
+				SetHandle (__r.Handle, JniHandleOwnership.TransferLocalRef);
+				_members.InstanceMethods.FinishCreateInstance (__id, this, null);
+			} finally {
+			}
+		}
+
+		static Delegate cb_getObject;
+#pragma warning disable 0169
+		static Delegate GetGetObjectHandler ()
+		{
+			if (cb_getObject == null)
+				cb_getObject = JNINativeWrapper.CreateDelegate ((Func<IntPtr, IntPtr, IntPtr>) n_GetObject);
+			return cb_getObject;
+		}
+
+		static IntPtr n_GetObject (IntPtr jnienv, IntPtr native__this)
+		{
+			global::Covariant.Returntypes.CovariantInterfaceImplementation __this = global::Java.Lang.Object.GetObject<global::Covariant.Returntypes.CovariantInterfaceImplementation> (jnienv, native__this, JniHandleOwnership.DoNotTransfer);
+			return JNIEnv.ToLocalJniHandle (__this.Object);
+		}
+#pragma warning restore 0169
+
+		public virtual unsafe global::Java.Lang.Object Object {
+			// Metadata.xml XPath method reference: path="/api/package[@name='covariant.returntypes']/class[@name='CovariantInterfaceImplementation']/method[@name='getObject' and count(parameter)=0]"
+			[Register ("getObject", "()Ljava/lang/Object;", "GetGetObjectHandler")]
```

Comment:
...and this raises a *different* yet related problem: the `[Register]` attribute is used as part of Java Callable Wrapper generation; the values present are used for the method overrides.

For example, if we do:

```csharp
public class Whatever : CovariantInterfaceImplementation {
    public override Java.Lang.Object Object {
        get {return this; /* mwa-ha-ha-ha-ha-ha! */}
    }
}
```

then the generated Java Callable Wrapper contain:

```java
public class Whatever extends CovariantInterfaceImplementation {
    public java.lang.Object getObject() {
        return n_getObject();
    }
    native java.lang.Object getObject();
}
```

This *will not compile*, as the Java `CovariantInterfaceImplementation` is:

```java
public class CovariantInterfaceImplementation {
    public String getObject() {/*...*/}
}
```

`Object getObject()` cannot override a `String getObject()` method.

What would *actually* need to happen is that the `[Register]` attribute needs to contain information for the *covariant override*:

```csharp
public virtual unsafe global::Java.Lang.Object Object {
    [Register ("getObject", "()Ljava/lang/String;", "GetGetObjectHandler" /* eh? */)]
    get { ... }
}
```

This will ensure that the Java Callable Wrapper is sane.

It *won't* fix the deliberate "think-o" in the above example: it's *supposed to be* a Java, but the C# compiler doesn't know that, and allows us to return a non-`String` value. This likely isn't fixable.

Question-to-which-I-don't-know-the-answer: is `GetGetObjectHandler` going to be correct here? I don't know.